File: fc-large1/ecommerce-app/src/main/java/com/fastcampus/helloecommerceservice/controller/dto/customer/CustomerDTO.java
package com.fastcampus.helloecommerceservice.controller.dto.customer;

import com.fastcampus.helloecommerceservice.domain.customer.Customer;
import lombok.Data;

@Data(staticConstructor = "of")
public class CustomerDTO {
    private final Long customerId;
    private final String customerName;
    private final String email;
    private final String address;
    private final String phoneNumber;

    public static CustomerDTO of(Customer customer) {
        return CustomerDTO.of(
                customer.getCustomerId(),
                customer.getName(),
                customer.getEmail(),
                customer.getAddress(),
                customer.getPhoneNumber());
    }
}

File: graphql/graphqlex/reactive-r2dbc/src/main/kotlin/com/yscorp/reactiver2dbc/dto/CustomerDto.kt
package com.yscorp.reactiver2dbc.dto

import com.yscorp.reactiver2dbc.domain.Customer

data class CustomerDto(
    val id: Int?,
    val name: String,
    val age: Int,
    val city: String,

    ) {

    constructor(customer: Customer) : this(
        id = customer.id,
        name = customer.name,
        age = customer.age,
        city = customer.city
    )


    fun toEntity(): Customer {
        return Customer(
            name = this.name,
            age = this.age,
            city = this.city
        )
    }
}
